src/Fmt_ast.ml

```diff
@@ -878,7 +884,7 @@ and fmt_core_type c ?(box = true) ?(in_type_declaration = false) ?pro
               fmt "< " $ row_fields rfs $ fmt " > "
               $ list ls "@ " (fmt "`" >$ str)
           | Open, Some _, _ -> impossible "not produced by parser" )
-        $ fits_breaks (if protect_token then " ]" else "]") "@ ]" )
+        $ fits_breaks (if protect_token then " ]" else "]") closing_break )
```

Comment:
minor nit: it would be more consistent style to change the definition of `closing_break` to compute a boolean that could be branched on here, in a similar way to how the long code of `protect_token` just computes a boolean but the effect on the formatting can be seen in context here.